scripts/core/editor3/html/to-html/HTMLGenerator.js

```diff
@@ -169,11 +171,53 @@ export class HTMLGenerator {
     /**
      * @ngdoc method
      * @name HTMLGenerator#html
+     * @param {bool} withAnnotations
      * @returns {string} HTML
      * @description Returns the HTML representation of the contentState stored by
      * this instance.
      */
-    html() {
-        return this.blocks.reduce((html, block, id) => html + this.convertBlock(block, id), '');
+    html(withAnnotations) {
+        if (withAnnotations) {
+            const highlights = this.blocks[0].getData();
+            const annotations = [];
+            let highlightIndex = 1;
+
+            // populate content state with entities
+            highlights.forEach((data, key) => {
+                const selection = JSON.parse(key);
+                const id = highlightIndex++;
+
+                const selectionState = SelectionState.createEmpty();
+                const contentStateWithEntity = this.contentState.createEntity(
+                    'ANNOTATION',
+                    'INMUTABLE',
+                    {id}
+                );
+
+                const entityKey = contentStateWithEntity.getLastCreatedEntityKey();
+
+                this.contentState = Modifier.applyEntity(
```

Comment:
This won't work because it may overwrite other entities (in our case links).

Reply:
yep, but not sure how it would work in the output anyhow. you can't have those 2 combined really, clicks would only work for 1

Reply:
Sure, but I think the suggestion in the review comment would work better, unless you see a problem with that. Also, if I remember correctly, the annotation gets triggered by clicking a small icon (I saw some design at some point) - so the icon would be separate from the link, making both available for clicking.

Reply:
at first I was handling it in the content block, but you can have annotations going over multiple blocks which makes it hard to handle given the rendering per block. so that's why it seemed much cleaner using the entity

Reply:
Yes, but the annotation HTML only needs to show up _at the end_ of the actual selection, so you only care about the last character in the annotation. It doesn't matter where it starts and whether it's the same block or another. Take a look [here](https://dev.sourcefabric.org/browse/SDFID-79), there only needs to be an icon at the end of the annotation selection, you don't need to make the whole text clickable.

Reply:
I am pointing to the screen shots in [SDFID-79](https://dev.sourcefabric.org/browse/SDFID-79). Please excuse me if anything has changed while I was away or if I'm missing something.

Reply:
I don't think this is in the requirements, the icon itself will be added by the other system, probably via css, so we need the `<span>`s around the annotated text

Reply:
Ok, that makes sense. But I still feel it's a bit unnatural to change the content state just for the sake of generating the HTML. We could still use the same pattern and open a span where an annotation starts and close it where it ends (or if it goes across blocks then when the block ends too).

I do understand that the implementation may be more complicated but it's more correct.

This is of course just my opinion so please feel free to chose whichever solution you like more...

Reply:
P.S. There is of course the link problem too. And who knows what other entities might show up in the future, so I'd advise against it.

Reply:
imo it's problematic anyhow if we allow the links and annotations to be mixed
because then we could have html like:
```
<span annotation-id="1">some <a href="">link</span> and link continues</a>
```
or vice versa

Reply:
That's true. That's not very semantic HTML. It would work though, nevertheless. I will let you decide here. To me it seems a bit hacky to create those entities.

Reply:
seems like this is not required (more info on superdesk-editorial slack channel)